File: gsw/GENERIC_ADCS/procedures/generic_adcs_test.rb
require 'cosmos'
require 'cosmos/script'
require 'generic_adcs_lib'

class GENERIC_ADCS_Functional_Test < Cosmos::Test
  def setup
    enable_adcs()
  end

  def test_algorithms
    enable_adcs()
    goto_sunsafe()
    goto_startracker_not_at_earth()
    goto_inertial()
  end

  def test_inputs
    adcs_confirm_css_data()
    adcs_confirm_imu_data()
    adcs_confirm_mag_data()
    # Go to inertial mode to ensure the star tracker is valid
    goto_inertial()
    adcs_confirm_st_data()
    # Go to sunsafe mode to ensure the FSS is pointed at the sun
    goto_sunsafe()
    adcs_confirm_fss_data()
  end
  
  def test_outputs
    adcs_confirm_rw_data()
  end

  def teardown
  end
end

class Generic_ADCS_Test < Cosmos::TestSuite
  def initialize
      super()
      add_test('GENERIC_ADCS_Functional_Test')
  end

  def setup
    enable_adcs()
  end
  
  def teardown
  end
end

File: gsw/GENERIC_ADCS/procedures/generic_adcs_test.py
import sys
import glob

for p in glob.glob('/gems/gems/openc3-cosmos-nos3-*/targets/GENERIC_ADCS/scripts'):
    if p not in sys.path:
        sys.path.append(p)

from openc3.script import cmd, tlm, check
from openc3.script.suite import Suite, Group

try:
    from nos3.generic_adcs_lib import safe_adcs, adcs_confirm_css_data, adcs_confirm_fss_data, adcs_confirm_imu_data, adcs_confirm_mag_data, adcs_confirm_st_data, adcs_confirm_rw_data
    from nos3.generic_adcs_app_test import run_generic_adcs_app_test 
except ImportError:
    pass

class GENERIC_ADCS_Functional_Test(Group):
    def setup(self):
        safe_adcs()

    def script_application(self):
        run_generic_adcs_app_test()

    def script_inputs(self):
        safe_adcs()
        adcs_confirm_css_data()
        safe_adcs()
        adcs_confirm_fss_data()
        safe_adcs()
        adcs_confirm_imu_data()
        safe_adcs()
        adcs_confirm_mag_data()
        safe_adcs()
        adcs_confirm_st_data()
        safe_adcs()

    def script_outputs(self):
        safe_adcs()
        adcs_confirm_rw_data()
        safe_adcs()

    def teardown(self):
        safe_adcs()

class Generic_adcs_Test(Suite):
    def __init__(self):
        super().__init__()
        self.add_group(GENERIC_ADCS_Functional_Test)

    def setup(self):
        safe_adcs()
  
    def teardown(self):
        safe_adcs()